### ISSUE TITLE
Idea: Allow whitespace between surname and given name?

### ISSUE THREAD

**Issue by SaltfishAmi** [birtles/rikaichamp#467]
Given text: `安倍 晋三`, as soon as we know that:
1. `安倍` is a surname, and is a longest match straight before the whitespace,
2. `晋` is valid Japanese character, and
3. Only whitespace ('` `' or '`　`') is separating the surname and the following valid Japanese character,

we can try to ignore the whitespace and see if there is a longer matching full name entry.

**Comment by SaltfishAmi** [comment]
This idea will not be able to solve this, though...

![Untitled](https://user-images.githubusercontent.com/14184974/101767045-a815f100-3b1e-11eb-8fc0-66eaa3a3e263.png)
(Every character is separated by unknown amount of whitespace)
But I think this is going a bit too far?

**Comment by birtles** [comment]
Yes, those latter cases appear quite a bit unfortunately (ideally those documents should simply be marked up using `letter-spacing` but oh well).

Also the code Rikaichamp uses to detect the range to lookup is already pretty complicated. Even just making it recognize years with half-width numbers is quite convoluted. Furthermore, that code runs on every single `mousemove` event so we need to be really careful about not making it slower.

For the original issue about handling names, I suspect that could be done without impacting performance too much.

**Comment by SaltfishAmi** [comment]
I found this somewhat similar to #255 .
